worker/include/Utils.hpp

```diff
@@ -418,16 +428,16 @@ namespace Utils
 #ifdef MS_MEM_POOL_FREE_ON_RETURN
 			std::free(ptr);
 #else
-			this->pool.push_back(ptr);
+			pool_data->push_back(ptr);
 #endif
 		}
-
-	private:
-		std::vector<T*> pool;
 	};
 
 	template<typename T>
 	thread_local Utils::ObjectPoolAllocator<T> Utils::ObjectPoolAllocator<T>::Pool;
+
+	template<typename T>
+	thread_local std::shared_ptr<std::vector<T*>> Utils::ObjectPoolAllocator<T>::pool_data;
```

Comment:
I don't think I understand why this is needed here in addition to definition above.